### ISSUE TITLE
Refactor testing setup

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#67]
Writing test cases for lint rules is a cumbersome at the moment.

We should refactor testing setup to be able to write tests easily, here are some things that would be useful:
- create testing fn for lint rule: 
  - `assert_lint_ok<T>(lint_rule: T, source_code)`
  - `assert_lint_err<T>(lint_rule: T, source_code) -> Vec<Diagnostic>`
- compare diagnostics partially - ie. assert that starts on certain column but ignore filename/line

**Comment by lucacasonato** [comment]
This is what I have right now: 

```rust
fn ban_ts_ignore() {
  let diagnostics = assert_lint_err::<BanTsIgnore>(
    r#"
// @ts-ignore
function foo() {
// pass
}

function bar() {
// @ts-ignore
const a = "bar";
}
  "#,
  );
  assert_has_diagnostic(&diagnostics, "banTsIgnore", 2, 0);
  assert_has_diagnostic(&diagnostics, "banTsIgnore", 8, 2);
}
```

Does that seem reasonable? I don't think we should be ignoring line number because it is actually important in many cases. I agree the file name can be ignored. Also should we also check the error message? I personally think the error code is enough.

**Comment by bartlomieju** [comment]
@lucacasonato looks good - I suggested skipping line number because in most cases tests can be written on single line:
```rust
assert_lint_ok::<RequireYield>("function foo() {}");
assert_lint_ok::<RequireYield>("function* bar() { yield "bar"; }");

assert_lint_err::<RequireYield>("function* bar() { return "bar"; }");
```
Maybe it would make sense to have a helper that asserts there is **single** diagnostics returned and separate helper that returns array of diagnostics (user gives number of expected diagnostics)?
```
assert_lint_err_n::<RequireYield>(3, "function* bar() { return "bar"; }");
```

**Comment by bartlomieju** [comment]
One more thought: asserting valid code can probably be done by passing vector of test cases
```rust
assert_lint_ok::<RequireYield>(vec![
  "function foo() {}",
  "function* bar() { yield "bar"; }",
]);
```

**Comment by bartlomieju** [comment]
Cleanly done by @lucacasonato in #72
